docs/ai_actions/extend_ai_actions.md

```diff
@@ -0,0 +1,370 @@
+```
+
+``` yaml
+[[= include_file('code_samples/ai_actions/config/services.yaml', 25, 28) =]]
+```
+
+Executing Actions this way has a major drawback: all the parameters are stored directly in the code and cannot be easily reused or changed.
+To manage configurations of an AI Action you need to use another concept: Action Configurations.
+
+## Action Configurations
+
+### Manage Action Configurations
+
+Action Configurations allow you to store the parameters for a given Action in the database and reuse them when needed.
+They can be managed [through the back office]([[= user_doc =]]/ai_actions/work_with_ai_actions.md), [data migrations](importing_data.md#ai-action-configurations), or through the PHP API.
+
+To manage Action Configurations through the PHP API, you need to use the [ActionConfigurationServiceInterface](../api/php_api/php_api_reference/classes/Ibexa-Contracts-ConnectorAi-ActionConfigurationServiceInterface.html) service.
+
+You can manage them using the following methods:
+
+- Creating them with `ActionConfigurationServiceInterface::createActionConfiguration()` by passing the [ActionConfigurationCreateStruct](../api/php_api/php_api_reference/classes/Ibexa-Contracts-ConnectorAi-ActionConfiguration-ActionConfigurationCreateStruct.html).
+- Updating them with `ActionConfigurationServiceInterface::updateActionConfiguration()` by passing the [ActionConfigurationUpdateStruct](../api/php_api/php_api_reference/classes/Ibexa-Contracts-ConnectorAi-ActionConfiguration-ActionConfigurationUpdateStruct.html).
+- Deleting them with `ActionConfigurationServiceInterface::deleteActionConfiguration()` by passing the [ActionConfigurationInterface](../api/php_api/php_api_reference/classes/Ibexa-Contracts-ConnectorAi-ActionConfigurationInterface.html).
+
+See the [AI Actions event reference](ai_action_events.md#action-configurations-management) for a list of events related to these operations.
+
+You can get a specific Action Configuration using the `ActionConfigurationServiceInterface::getActionConfiguration()` method and search for them using the `ActionConfigurationServiceInterface::findActionConfigurations()` method.
+See [Action Configuration Search Criteria reference](action_configuration_criteria.md) and [Action Configuration Search Sort Clauses reference](action_configuration_sort_clauses.md) to discover query possibilities.
+
+The following example creates a new Action Configuration:
+
+``` php hl_lines="3 17"
+[[= include_file('code_samples/ai_actions/src/Command/ActionConfigurationCreateCommand.php', 59, 76) =]]
+```
+
+Actions Configurations are tied to a specific Action Type and are translatable.
+
+### Execute Actions with Action Configurations
+
+Reuse existing Action Configurations to simplify the execution of AI Actions.
+You can pass one directly to the `ActionServiceInterface::execute()` method:
+
+``` php hl_lines="7-8"
+[[= include_file('code_samples/ai_actions/src/Command/ActionConfigurationCreateCommand.php', 77, 85) =]]
```

Comment:
I would have prefer a rewrite of the previous example.

```suggestion
        $action = new GenerateAltTextAction(new Image([$imageEncodedInBase64]));

        $actionConfiguration = $this->actionConfigurationService->getActionConfiguration('TODO_alt_text');

        $output = $this->actionService->execute($action, $actionConfiguration)->getOutput();
```

But no big deal, reader should connect the dot and see how this could have been used in the previous example.